allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -91,7 +91,11 @@ def tokens_to_indices(
                     f" for the following token: {token.text}"
                 )
 
-        return {index_name: indices}
+        # The mask has 1 for real tokens and 0 for padding tokens. Only real
+        # tokens are attended to.
+        attention_mask = [1] * len(indices)
+
+        return {index_name: indices, "attention_mask": attention_mask}
```

Comment:
At https://github.com/allenai/allennlp/blob/master/allennlp/nn/util.py#L626 (and apparently many other places), we assume that the mask is stored under the key `"mask"`.

Reply:
We also assume it's a `LongTensor`, not a `List`.

Reply:
The `tokens_to_indices` method is expected to return `Dict [str, List [int]]`, so I think it's better not to return a `LongTensor`.  Is it correct?

Reply:
Yes, @RyujiTamaki, you're right.  The conversion to a tensor happens in `as_padded_tensor`, not here.